imap_processing/tests/codice/test_codice_l1a.py

```diff
@@ -203,30 +203,34 @@ def test_l1a_data_array_values(test_l1a_data: xr.Dataset, index):
 
     descriptor = DESCRIPTORS[index]
 
-    # Mark currently broken/unsupported datasets as expected to fail
-    # TODO: Remove these once they are supported
-    if index in [0, 1, 15, 16, 17]:
-        pytest.xfail("Data product is currently unsupported")
-
-    # TODO: Currently only lo-(n)sw-angular data can be validated, expand this
-    #       to other data products as I validate them.
-    if descriptor in ["lo-sw-angular", "lo-nsw-angular"]:
+    # TODO: Currently only the following products can be validated, expand this
+    #       to other data products as I can validate them.
+    able_to_be_validated = ["lo-sw-angular", "lo-nsw-angular"]
+    if descriptor in able_to_be_validated:
         counters = getattr(
             constants, f'{descriptor.upper().replace("-","_")}_VARIABLE_NAMES'
         )
         processed_dataset = test_l1a_data[index]
         validation_dataset = load_cdf(VALIDATION_DATA[index])
 
-        # Joey says that the shape of the data arrays (i.e. how they are
-        # arranged) does not need to follow a specific order, and the SDC can
-        # decide to arrange them how we see fit. As such, the shape of the CDFs
-        # that SDC produces may not match the validation data. To get around
-        # this, compare the sum of the values of the data arrays.
         for counter in counters:
+            # Ensure the data array values are equal by calculating the sum of
+            # each and comparing the result
             np.testing.assert_array_equal(
-                getattr(processed_dataset, counter).data.sum(),
-                getattr(validation_dataset, counter).data.sum(),
+                processed_dataset[counter].data.sum(),
+                validation_dataset[counter].data.sum(),
+            )
+
+            # Ensure the data array shapes are equal
+            assert (
+                processed_dataset[counter].data.shape
+                == validation_dataset[counter].data.shape
             )
+
+            # TODO: Once Joey and I figure out some small discrepancies with
+            #       some data products, we may be able to just compare the data
+            #       arrays directly (i.e. assert_array_equal on just the .data
+            #       attribute, instead of sum or shape.
```

Comment:
This doesn't make sense to me that the sum would be good but the array comparison not. Are you just getting very minimal differences or a 1 and -1 difference that cancel out in the sum?

There is also `xr.testing.assert_equal()` for comparing the datasets directly that I saw someone else using too.

Reply:
Using `xr.testing.assert_equal()` on the entirety of the data arrays is what I am striving towards. For now, I split up the checks by `sum()` and `shape()` because i'm still sorting out some value differences in some of the data products with Joey. So it has been useful for me to easily identify issues. 

But yes, eventually I would like this test function to just compare the arrays, which should prove that I am producing valid data.

Reply:
My curiosity still stands though. How can your sum be valid but not the individual values? Do you really want an `assert_almost_equal()` instead of exactly equal?

Reply:
Currently for some data products (not listed in this test because they would break, but I've been playing around with them locally) have the correct values but differ in shape from the validation data. I am still sorting out some of these differences with Joey. So basically, this is just a workaround / playground for testing as I am sorting these things out, hence why I put the `TODO` note there -- eventually this will all get moved to just comparing the data arrays with `assert_equal()`.

Reply:
> not listed in this test because they would break, but I've been playing around with them locally

My preference would be to leave these as the real test (not separate sum and shape assertions) for now then if they do pass. The testing assert_array_equal should tell you whether the shapes, specific elements etc are different in your local debugging, so should be nicer to debug output with too.